Jellyfin.Api/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,138 @@
+#nullable enable
+#pragma warning disable CA1801
+#pragma warning disable SA1313
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Jellyfin.Api.Models.NotificationDtos;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Notifications;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Notifications;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers
+{
+    /// <summary>
+    /// The notification controller.
+    /// </summary>
+    public class NotificationsController : BaseJellyfinApiController
+    {
+        private readonly INotificationManager _notificationManager;
+        private readonly IUserManager _userManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NotificationsController" /> class.
+        /// </summary>
+        /// <param name="notificationManager">The notification manager.</param>
+        /// <param name="userManager">The user manager.</param>
+        public NotificationsController(INotificationManager notificationManager, IUserManager userManager)
+        {
+            _notificationManager = notificationManager;
+            _userManager = userManager;
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notifications.
+        /// </summary>
+        /// <param name="UserID">The UserID.</param>
+        /// <param name="IsRead">An optional filter by IsRead.</param>
+        /// <param name="StartIndex">The optional index to start at. All notifications with a lower index will be dropped from the results.</param>
+        /// <param name="Limit">An optional limit on the number of notifications returned.</param>
+        /// <returns>A read-only list of all of the user's notifications.</returns>
+        [HttpGet("{UserID}")]
+        public IReadOnlyList<NotificationDto> GetNotifications(
+            [FromRoute] string UserID,
+            [FromQuery] bool? IsRead,
+            [FromQuery] int? StartIndex,
+            [FromQuery] int? Limit)
```

Comment:
@Bond-009 this framework was generating the endpoint parameters from these variable names, so the new plan was to use title casing here. Any issues with that? I know it's not ideal for local variables, but I'd rather have a new web framework than correct casing in this situation.

Reply:
> @Bond-009 this framework was generating the endpoint parameters from these variable names, so the new plan was to use title casing here. Any issues with that? I know it's not ideal for local variables, but I'd rather have a new web framework than correct casing in this situation.

Actually, I'll be pushing a new commit shortly reverting this. My changes over in #2848 will ensure the response from DTO's being serialized retains title casing, but the local variables being camel case will still bind to title cased form & query parameters sent from the frontend.